### DISCUSSION TITLE
ModuleNotFoundError: No module named 'phonemizer'

### DISCUSSION THREAD

**Discussion by MrBibbles3** [nerdaxic/glados-voice-assistant#40] [Q&A]
I am running Ubuntu on windows using wsl2 I think it's called?
I have looked in the file it is specifying and it is right, there is no phonemizer.py file in my download. Did I miss downloading something? Where is phonemizer.py?
![image](https://github.com/nerdaxic/glados-voice-assistant/assets/150344334/5620bc67-fc70-403d-be6d-b9b63bcd9ebb)

**Comment by MrBibbles3** [comment]
UPDATE: I commented the line "from phonemizer..." to see any other issues, and I immediately got another issue from numbers.py trying to "import inflect". Is there something I haven't installed properly that all of these belong to?

**Reply by eternalliving** [reply]
Try running glados.py from glados-voice-assistant and not the glados-tts directory. 

Are you wanted to just run the tts engine, or the whole voice assistant?

**Reply by MrBibbles3** [reply]
Thanks for replying! I have HomeAssistant running on a Raspberry Pi and use my Alexa to voice commands. Since I do not have a background in programming, I didn't think I'd be able to set up the glados tts in HomeAssistant (I saw the comment that the Pi's CPU can't run it) so I am just wanting to make audio files using the tts and have alexa play them using a routine (which I know how to do). TLDR: I want to make my own tts Glados audio files and use HA to play them.

I tried running the glados.py from glados-voice-assistant and got the message: ModuleNotFoundError: No module named 'pyaudio'.
![image](https://github.com/nerdaxic/glados-voice-assistant/assets/150344334/796b8256-a7a9-4d97-886a-8342626daf00)

**Reply by eternalliving** [reply]
If you just want to use the tts engine you could install it on a more powerful laptop and generate all the lines you think you'll need. I believe there are also a couple of online tts engines that might still be working, if not, I might be able to spin one up for you. 

Someone has managed to get the tts engine working on the pi, but from what I understand, it takes about 15-20s to generate the audio. My laptop gpu does it in 20ms!

If you think you'll only need pregenerated audio, and not the ability to make new audio on demand, try https://glados.c-net.org

**Reply by MrBibbles3** [reply]
Thanks, yeah  I have been using the website for the time being. However, I would like to learn how to use it with Home Assitant so I can use GLADoS all the time. Using the mp3's as responses has limits as when I ask the time my alexa of course uses her voice instead sigh. I have currently set up a very large amount of routines activated by variations of "turn this thing on", that then plays a premade tts mp3 response I uploaded. Which is good for the moment, but not my end goal.

If it helps, I have an old gaming pc I now use as a server sometimes. Could I run the tts engine through that, then link it with Home Assistant, so Alexa can generate responses on the fly instead of having to import premade mp3s? I know some basic coding from using Arduino and playing around with YAML used by Home Assistant, but unsure how to get the tts engine working (because of the errors in my last 2 comments). (btw the server pc has gen 9 i7 I believe and 16GB RAM with a GTX1080, so it could easily handle anything).

**Reply by eternalliving** [reply]
It's definitely nicer to have on the fly tts, otherwise it can feel a little repetitive. 

I haven't use the tts engine with HA, but I believe it is pretty easy to link in. If you're running the tts engine on a windows machine there are a few extra steps to get it installed, but it works great, and if you can use the GPU, even better. I have a small script on my rpi that sends a tts query to my laptop, then plays back the result on the rpi.

See if you can get the TTS engine installed on the old gaming machine and let me know if you run into issues. There are a few posts under the closed issues that might help. This is one of them: https://github.com/nerdaxic/glados-voice-assistant/issues/24
